src/components/TierPurchase/TierPurchase.js

```diff
@@ -12,142 +12,104 @@ import { func, instanceOf, node, number, string } from 'prop-types'
 import FarmhandContext from '../Farmhand/Farmhand.context'
 import './TierPurchase.sass'
 
-export class TierPurchase extends Component {
-  state = {
-    selectedTier: '',
-  }
+export function TierPurchase({
+  description,
+  handleTierPurchase,
+  maxedOutPlaceholder,
+  money,
+  purchasedTier,
+  renderTierLabel,
+  tiers,
+  title,
+}) {
+  const [selectedTier, setSelectedTier] = useState(purchasedTier)
 
-  tierValues = [...this.props.tiers.entries()]
+  const tierValues = [...tiers.entries()]
 
-  get selectedTierNumber() {
-    return Number(this.state.selectedTier)
-  }
+  const selectedTierNumber = Number(selectedTier)
+
+  const hasPurchasedTier = tierLevel => tierLevel <= purchasedTier
+  const hasPurchasedHighestTier = hasPurchasedTier(tierValues.slice(-1)[0][0])
+
+  if (hasPurchasedTier(selectedTier)) {
+    const nextTierNumber = selectedTierNumber + 1
+    const nextTierToPurchase = tiers.get(nextTierNumber)
 
-  get canPlayerBuySelectedTier() {
-    const {
-      props: { money },
-      selectedTierNumber,
-    } = this
+    if (nextTierToPurchase && money >= nextTierToPurchase.price) {
+      setSelectedTier(`${nextTierNumber}`)
+    }
+  }
 
-    const selectedTier = this.props.tiers.get(selectedTierNumber)
+  const canPlayerBuySelectedTier = () => {
+    const selectedTier = tiers.get(selectedTierNumber)
 
     return (
       !!selectedTier &&
-      !this.hasPurchasedTier(selectedTierNumber) &&
+      !hasPurchasedTier(selectedTierNumber) &&
       money >= selectedTier.price
     )
   }
 
-  hasPurchasedTier = tierLevel => tierLevel <= this.props.purchasedTier
+  const onClickBuy = () => {
+    const canAfford = tiers.get(selectedTierNumber).price <= money
 
-  handleTierPurchase = () => {
-    const {
-      props: { handleTierPurchase, money },
-      selectedTierNumber,
-    } = this
-    const { price } = this.props.tiers.get(selectedTierNumber)
-
-    if (money >= price) {
+    if (canAfford) {
       handleTierPurchase(selectedTierNumber)
     }
   }
 
-  onSelectChange = ({ target: { value } }) => {
-    this.setState({ selectedTier: value })
-  }
-
-  updateSelectedTier = () => {
-    const {
-      props: { money },
-      selectedTierNumber,
-    } = this
-
-    const nextTierNumber = selectedTierNumber + 1
-    const nextTierToPurchase = this.props.tiers.get(nextTierNumber)
-
-    if (nextTierToPurchase && money >= nextTierToPurchase.price) {
-      this.setState({ selectedTier: String(nextTierNumber) })
-    }
-  }
-
-  componentDidMount() {
-    this.updateSelectedTier()
-  }
-
-  componentDidUpdate(prevProps) {
-    if (this.props.money !== prevProps.money) {
-      this.updateSelectedTier()
-    }
+  const onSelectChange = ({ target: { value } }) => {
```

Comment:
I realize that given my event handler name comment above, this component is self-inconsistent. Sorry about that... 😅

I'd rename this to `handleSelectChange`, but this is preexisting so I don't consider it a necessary change for this PR.

Reply:
My understanding of [the article](https://jaketrent.com/post/naming-event-handlers-react) is that handlers exposed as props (the public API) should be named `on*`, and private internal handlers should be named `handle*`. Since this is the latter, `handleSelectChange` seems to conform to what the article proposes. I think I have that right?